LayoutTests/editing/selection/ios/bidi-visually-contiguous-selection.html

```diff
@@ -0,0 +1,54 @@
+<!DOCTYPE html> <!-- webkit-test-runner [ useFlexibleViewport=true VisuallyContiguousBidiTextSelectionEnabled=true ] -->
+<html>
+<head>
+<meta name="viewport" content="width=device-width, initial-scale=1">
+<meta charset="utf-8">
+<script src="../../../resources/ui-helper.js"></script>
+<script src="../../../resources/js-test.js"></script>
+<style>
+body, html {
+    font-size: 20px;
+    font-family: system-ui;
+}
+
+.start {
+    border: 1px solid tomato;
+    padding: 4px;
+}
+
+.stop {
+    border: 1px solid orange;
+    padding: 4px;
+}
+</style>
+<script>
+jsTestIsAsync = true;
+
+addEventListener("load", async () => {
+    description("This test verifies that the text selection appears visually contiguous when selecting across bidi text boundaries.");
+
+    await UIHelper.longPressElement(document.querySelector(".start"));
+    await UIHelper.waitForSelectionToAppear();
+
+    const touchStart = UIHelper.midPointOfRect(await UIHelper.getSelectionStartGrabberViewRect());
+    const touchEnd = UIHelper.midPointOfRect(document.querySelector(".stop").getBoundingClientRect());
+
+    await UIHelper.sendEventStream(new UIHelper.EventStreamBuilder()
+        .begin(touchStart.x, touchStart.y)
+        .move(touchEnd.x, touchEnd.y, 0.5)
+        .takeResult());
+
+    selectionRectsBeforeEndingSelection = await UIHelper.getUISelectionViewRects();
+    await UIHelper.sendEventStream(new UIHelper.EventStreamBuilder().end().takeResult());
+
+    shouldBe("selectionRectsBeforeEndingSelection.length", "1");
+    finishJSTest();
+});
+</script>
+</head>
+<body>
+    <p>Arabic, <span class="stop">مثل</span> هذا النص, is right to <span class="start">left</span>.</p>
```

Comment:
Can we add a ref test which asserts that the selections are visually contiguous?
Also, what happens to selection set by scripts? We don't show them visually contiguous?

Reply:
> Can we add a ref test which asserts that the selections are visually contiguous?

Hm... a ref test is tricky since selections don't render as part of web content (and thus, don't appear in the reference snapshots). However, I think I can make a ref test work in one of two ways:
1. By adding some JavaScript helper to check for visual discontinuity, and append a simple `"Visually contiguous"` or `"Visually discontiguous"` text to the page, or:
2. By adding some JavaScript helper to append absolutely-positioned DOM elements to the page over the selection rects.

> Also, what happens to selection set by scripts? We don't show them visually contiguous?

Right, we'd want to preserve existing behavior and just fall back to a visually discontiguous (but logically contiguous) selection. I haven't gotten to implementing that behavior yet — it'll be in a later patch.

Reply:
For now, I think I'll implement the helper function in (1) inside `ui-helper.js`, and use it here so that the expectation is something like:

```
PASS selectionIsVisuallyContiguous is true
```